internal/storage/db/saveDB.go

```diff
@@ -6,16 +6,22 @@ import (
 	"github.com/kamencov/go-musthave-shortener-tpl/internal/utils"
 )
 
-func (p *PstStorage) SaveURL(shortURL, originalURL string) error {
+func (p *PstStorage) SaveURL(shortURL, originalURL, userID string) error {
+	var user interface{}
+	if userID == "" {
+		user = nil
+	} else {
+		user = userID
+	}
```

Comment:
Все-таки не стоит использовать `interface{}` там, где известен тип переменной. Если необходимо чтобы строка была nullable, можно использовать указатель.
```suggestion
	var user *string
	if userID != "" {
		user = userID
	}
```

Reply:
именно так не работает, сработало &userID